vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -145,6 +170,23 @@ public ISpaceLinkDisplay createSpaceLink(String title, ISpace source, float posi
         return display;
     }
 
+    @Override
+    public IExternalLinkDisplay createExternalLink(String title, ISpace space, float positionX, float positionY,
+            String url) throws SpaceDoesNotExistException {
```

Comment:
this exception is never thrown. also, looking at my code above, there would be a different exception thrown if the space does not exist. this also needs to be fixed.

Reply:
Ok. Here we don't have to check for target space so I can remove that exception. Should we check if the URL(externallink) is a valid one or not and throw an Exception?